### PR TITLE
add sorting for the members page

### LINKED ISSUE #1941: add sorting options to the members page
We want to have some basic sorting options for viewing the members in the channel. I *think* the options we want to support so far are sorting alphabetically on name and sorting by the date the user joined the channel.

#### Acceptance Criteria:

- [x] add name sorting
- [x] add channel join date to the elasticsearch index
- [x] add options to sort on that date

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #1941 

#### What's this PR do?

this adds sorting to the channel members page, currently it just lets you pick between sorting based on the members' names and when they joined the channel.

#### How should this be manually tested?

If you go to the channel member page you should see options for switching the sort. it should change it in a way that makes sense - so the people should be alphabetically sorted with the name option, and the people should be sorted so that the most recently added to the channel are at the top with the channel join date option.

You'll need to run the `recreate_index` command to put the new info into the index.

#### Screenshots (if appropriate)

asdffffffffjfjfjfffasffdfa
asdffffffffjfjfjfffas
asdffffffffjfjfj
asdfffffff

### CODE REVIEW

**Comment by Ferdi** [comment]
sort by:  New, Name

File: static/js/containers/ChannelMembersPage.js [hunk 1]
```diff
@@ -6,19 +6,42 @@ import { connect } from "react-redux"
...
 
 import type { SearchParams, Result } from "../flow/searchTypes"
 
 export const CHANNEL_MEMBERS_PAGE_SIZE = 50
 
-const channelMemberParams = (channelName, from) => ({
+export const getSortOption = (sortField: string, channelName: string) => {
+  switch (sortField) {
+  case MEMBERS_SORT_AUTHOR_NAME:
+    return "asc"
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/js/containers/ChannelMembersPage.js]: I'm not sure - @Ferdi and @pdpinch what do y'all think?

File: profiles/api.py [hunk 2]
```diff
@@ -47,6 +47,31 @@ def get_channels(user):
...
+    Args:
+        user(django.contrib.auth.models.User): the user to retrieve channel names for
+
+    Returns:
+        set of (str: Channel names, datetime: when they joined)
+    """
+    names_and_dates = list(
+        user.channelsubscription_set.values_list("channel__name", "created_on")
+    ) + list(
+        ChannelGroupRole.objects.filter(group__in=user.groups.all()).values_list(
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply profiles/api.py]: yeah, I wasn't sure the best way to go about this. I wanted to make sure that any moderators who aren't subscribed would still show up on the members page.

in the case that a moderator is also a subscriber, the code I have right now will keep whichever is the oldest date. do you think that approach makes sense?

[the remark this answers is not in the document]
Comment (mbertrand) [reply profiles/api.py]: 👍

File: static/js/containers/ChannelMembersPage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ferdi) [reply static/js/containers/ChannelMembersPage.js]: I'd not complicate it.

**Comment by aliceriot** [comment]
@mbertrand I made that tweak to the test

**Pull request merged**